daemon/sdnotify_test.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2016 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package daemon
+
+import (
+	"os"
+	"testing"
+)
+
+// TestDaemon
+func TestDaemon(t *testing.T) {
+	// (true, nil) - notification supported, data has been sent
+	// this test should work as long as the test host runs systmed
+	os.Setenv("NOTIFY_SOCKET", "/run/systemd/notify")
```

Comment:
> we avoid sending fake events to the real systemd during the test.

Because `SdNotify()` has hardcoded `NOTIFY_SOCKET` we need to change its value for the test. This will break the notification system (if exist) for a the time of the test. So we have 2 solutions here:

1) If `NOTIFY_SOCKET` exists we use it and systemd receives the test notifications, I do not think that it is a problem as long as the notifications do not carry a message that can be interpreted as command (e.g. READY=1, STATUS=blablabla). If `NOTIFY_SOCKET` do not exist we create it.

2) We need to change `SdNotify()` in a way that can be specified a different socket `func SdNotify(notifySocket, state string) (sent bool, err error) `. And have these cases:
* notifySocket == nil --> Notification non supported.
* notifySocket == "" --> Use the value in `NOTIFY_SOCKET` to get the socket.
* notifySocket == "VARIABLE_1" --> Use the value in `VARIABLE_1` to get the socket.

I am for the first solution, I do not think that if systemd receives some bogus notifications it is a problem.